test/utils

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+set -u
+
+cd "$( dirname "${BASH_SOURCE[0]}" )" || return
+
+export LANG=C LC_ALL=C
+
+function teardown() {
+    EXIT_CODE="$(tr -d 0 < "$OUTPUT_DIR/EXIT_CODE.txt")"
+    rm -r "$INPUT_DIR" "$OUTPUT_DIR"
+    exit "${EXIT_CODE:-0}"
+}
+
+INPUT_DIR="$(mktemp -d)"
+OUTPUT_DIR="$(mktemp -d)"
+trap "teardown" EXIT
+
+function comment() {
+    sed -e 's/^/# /'
+}
+
+function run_test() {
+    local test_name="$1"
+    local command_name="$2"
+    shift 2
+
+    (
```

Comment:
I think calling bash in a sub-shell here is superfluous.

Reply:
It is definitely not strictly necessary, but I did it anyways, because I think that it's marginally better/safer to run tests in a forked subshell, in case that child process segfaults, corrupts the environment and what not. If you don't have any strong argument against it, I would leave it this way.